compiler/rustc_data_structures/src/sharded.rs

```diff
@@ -1,31 +1,26 @@
 use crate::fx::{FxHashMap, FxHasher};
 #[cfg(parallel_compiler)]
-use crate::sync::is_dyn_thread_safe;
-use crate::sync::{CacheAligned, Lock, LockGuard};
+use crate::sync::{is_dyn_thread_safe, CacheAligned};
+use crate::sync::{Lock, LockGuard};
 use std::borrow::Borrow;
 use std::collections::hash_map::RawEntryMut;
 use std::hash::{Hash, Hasher};
 use std::mem;
 
-#[cfg(parallel_compiler)]
 // 32 shards is sufficient to reduce contention on an 8-core Ryzen 7 1700,
 // but this should be tested on higher core count CPUs. How the `Sharded` type gets used
 // may also affect the ideal number of shards.
 const SHARD_BITS: usize = 5;
```

Comment:
This is still used by `get_shard_hash`. Should it keep both possible values? Or should `get_shard_hash` be refactored out?

Reply:
`get_shard_hash` isn't actually used for anything with `cfg(not(parallel_compiler))`.

Reply:
Then, should it be `cfg(parallel_compiler)` to avoid accidental use?

Reply:
The function is still used with `cfg(not(parallel_compiler))`, but it's value doesn't end up used with optimizations applied as `get_shard_by_index` ignores the hash then.